charts/osm-arc/templates/validate-single-mesh.yml

```diff
@@ -0,0 +1,154 @@
+  annotations:
+    "helm.sh/hook-weight": "15"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+subjects:
+- kind: ServiceAccount
+  name: validate-single-mesh-account
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: validate-single-mesh-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: validate-single-mesh
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook-weight": "20"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeededd
+spec:
+  ttlSecondsAfterFinished: 0
+  backoffLimit: 5
+  template:
+    metadata:
+      labels:
+        app: validate-single-mesh
+    spec:
+      serviceAccountName: validate-single-mesh-account
+      automountServiceAccountToken: true
+      restartPolicy: Never
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: validate-single-mesh-aks-addon
+          image: {{ .Values.kubectl.image.name }}:{{ .Values.kubectl.image.tag }}
+          imagePullPolicy: IfNotPresent
+          env: 
+            - name: enforcesinglemesh
+              value: {{ .Values.osm.OpenServiceMesh.enforceSingleMesh | quote }}
+            - name: validatenamespaces
+              value: {{ .Values.OpenServiceMesh.validateNamespaces }}
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              osm_namespaces=$(kubectl get deployments -l app=osm-controller -A -o=jsonpath={..metadata.namespace})
+              for namespace in $osm_namespaces
+              do
+                for validateNamespace in ${validatenamespaces} 
+                do
+                  if [ "$namespace" = "kube-system" ]; then
+                    if [ "${enforcesinglemesh}" = "true" ]; then
+                      echo "osm already exists in namespace $namespace. osm-arc enforces single-mesh cluster."
+                      exit 1
+                    else
+                      aks_enforce_single_mesh=$(kubectl get deployment -l app=osm-controller -n kube-system -o jsonpath={.items..metadata.labels..enforceSingleMesh})
+                      if [ "$aks_enforce_single_mesh" = "true" ]; then
+                        echo "osm in namespace $namespace already exists and enforces single-mesh cluster."
+                        exit 1
+                      fi
+                    fi
+                  fi 
+                done
+              done
+        - name: validate-single-mesh-oss
+          image: {{ .Values.helm.image.name }}:{{ .Values.helm.image.tag }}
+          imagePullPolicy: IfNotPresent
+          env: 
+            - name: enforcesinglemesh
+              value: {{ .Values.osm.OpenServiceMesh.enforceSingleMesh | quote }}
+            - name: releasenamespace
+              value: {{ .Release.Namespace }}
+          command: 
+            - "/bin/sh"
```

Comment:
@nshankar13 I worry about having to maintain one more bash script, which could make or break an OSM installation.
We should have similar functionality in the OSM CLI (go code).
Do you think it is reasonable to replace the `bash` here with a container, which has the same checks implemented in Go?

I'd propose we
 - [ ] containerize the OSM CLI
 - [ ] publish the new container
 - [ ] use said container here with the right (perhaps new) OSM CLI command to check for this

Reply:
@draychev what exactly do you mean by make or break an OSM installation? Provided that the AKS add-on is always in the kube-system namespace, and the upstream/arc chart names remain as osm/osm-arc, I can't foresee any subsequent updates to the pre-install validation job.

Reply:
Also, the validation for arc's enforce-single-mesh works a bit differently than how upstream does it, since for osm-arc we want to allow re-installation in the same namespace to update the version (see lines 136 - 139 in validate-single-mesh.yml in this PR).

Reply:
@draychev the OSM CLI checks this on install. I've actually wondered if we should be moving this logic to the osm controller itself with some sort of webhook so that it would be uniform for upstream and downstream

OSM CLI Code: https://github.com/openservicemesh/osm/blob/main/cmd/cli/install.go#L217-L230